components/gif_picker/components/InfiniteScroll/index.tsx

```diff
@@ -0,0 +1,187 @@
+    defaultLoader: React.ReactNode = null;
+    pageLoaded = 0;
+
+    static defaultProps: InfiniteScrollDefaultProps = {
+        element: 'div' as keyof JSX.IntrinsicElements,
+        hasMore: false,
+        initialLoad: true,
+        pageStart: 0,
+        threshold: 250,
+        useWindow: true,
+        isReverse: false,
+        containerHeight: null,
+        scrollPosition: null,
+    }
+
+    componentDidMount() {
+        this.pageLoaded = this.props.pageStart ?? 0;
+        this.attachScrollListener();
+        this.setScrollPosition();
+    }
+
+    componentDidUpdate() {
+        this.attachScrollListener();
+    }
+
+    render() {
+        const {
+            children,
+            element,
+            hasMore,
+            initialLoad, // eslint-disable-line @typescript-eslint/no-unused-vars
+            loader,
+            loadMore, // eslint-disable-line @typescript-eslint/no-unused-vars
+            pageStart, // eslint-disable-line @typescript-eslint/no-unused-vars
+            threshold, // eslint-disable-line @typescript-eslint/no-unused-vars
+            useWindow, // eslint-disable-line @typescript-eslint/no-unused-vars
+            isReverse, // eslint-disable-line @typescript-eslint/no-unused-vars
+            scrollPosition, // eslint-disable-line @typescript-eslint/no-unused-vars
+            containerHeight,
+            ...rest
+        } = this.props;
+
+        const baseElementProps: InfiniteScrollElementProps = {
+            ref: (node) => {
+                this.scrollComponent = node;
+            },
+            ...rest,
+        };
+
+        const elementProps = containerHeight ? {...baseElementProps, style: {height: containerHeight}} : baseElementProps;
+
+        return React.createElement(element, elementProps, children, hasMore && (loader || this.defaultLoader));
+    }
+
+    calculateTopPosition(el: HTMLElement | Element | null | undefined): number {
+        if (!isHtmlElement(el)) {
+            return 0;
+        }
+
+        return el.offsetTop + this.calculateTopPosition(el.offsetParent);
+    }
+
+    setScrollPosition() {
+        const {scrollPosition} = this.props;
+        if (scrollPosition != null) {
+            window.scrollTo(0, scrollPosition);
+        }
+    }
+
+    scrollListener = () => {
+        const el = this.scrollComponent;
+        const scrollEl = window;
+
+        if (!el) {
+            return;
+        }
+
+        let offset = 0;
+        if (this.props.useWindow) {
+            const scrollTop = ('pageYOffset' in scrollEl) ? scrollEl.pageYOffset : (document.documentElement || document.body.parentNode || document.body).scrollTop;
+            if (this.props.isReverse) {
+                offset = scrollTop;
+            } else {
+                offset = this.calculateTopPosition(el) + (el.offsetHeight - scrollTop - window.innerHeight);
+            }
+        } else if (this.props.isReverse) {
+            offset = el.parentElement?.scrollTop ?? 0;
```

Comment:
I changed `parentNode` in favour of `parentElement`. The return type of `parentNode` is incompatible because `Node` is a base type that `HTMLElement` implements. We know that at run time this is truthy but TS at compile time not. So instead narrow the type through casting, I'm using `parentElement` instead.